packages/amplify-provider-awscloudformation/lib/amplify-service-manager.js

```diff
@@ -0,0 +1,344 @@
+const fs = require('fs-extra');
+const path = require('path');
+const inquirer = require('inquirer');
+const sequential = require('promise-sequential');
+const S3 = require('../src/aws-utils/aws-s3');
+const { getConfiguredAmplifyClient } = require('../src/aws-utils/aws-amplify');
+const constants = require('./constants');
+
+async function init(amplifyServiceParams) {
+  const { context, awsConfig, projectName, envName, stackName } = amplifyServiceParams;
+
+  const amplifyClient = await getConfiguredAmplifyClient(context, awsConfig);
+
+  let amplifyAppId;
+
+  if (context.exeInfo && context.exeInfo.inputParams && context.exeInfo.inputParams.amplify && context.exeInfo.inputParams.amplify.appId) {
+    const inputAmplifyAppId = context.exeInfo.inputParams.amplify.appId;
+    try {
+      const getAppResult = await amplifyClient
+        .getApp({
+          appId: inputAmplifyAppId,
+        })
+        .promise();
+      context.print.info(`Input Amplify appId is verified: ${inputAmplifyAppId}`);
+      context.print.info(`Amplify App name is: ${getAppResult.app.name}`);
+      amplifyAppId = inputAmplifyAppId;
+    } catch (e) {
+      context.print.error(`Input Amplify appId is invalild: ${inputAmplifyAppId}`);
+      context.print.info(e);
+      throw e;
+    }
+  }
+
+  if (!amplifyAppId) {
+    // in the "amplify env add" workflow, there might be other envs, new env can be added to existing appId
+    let teamProviderFilePath;
+    try {
+      teamProviderFilePath = context.amplify.pathManager.getProviderInfoFilePath();
+    } catch (e) {
+      // do nothing
+    }
+
+    if (teamProviderFilePath && fs.existsSync(teamProviderFilePath)) {
+      const teamProviderInfo = context.amplify.readJsonFile(teamProviderFilePath);
+
+      const envList = Object.keys(teamProviderInfo);
+
+      let appIdsInTheSameLocalProjectAndRegion = [];
+      for (let i = 0; i < envList.length; i++) {
```

Comment:
```for (const env of Object.keys(teamProviderInfo))```

Reply:
lol, I've heard JS developers don't like C style for loop, and I got one here.